storm-core/src/jvm/backtype/storm/blobstore/BlobStoreUtils.java

```diff
@@ -0,0 +1,256 @@
+                // there is a subsequent update and delete, the non-leader
+                // nimbodes might throw an exception.
+                LOG.info("KeyNotFoundException Key: {} {}", key, knf);
+            } catch (Exception exp) {
+                // Logging an exception while client is connecting
+                LOG.error("Exception {}", exp);
+            }
+        }
+
+        if (!isSuccess) {
+            LOG.error("Could not download blob with key" + key);
+            return false;
+        }
+        return isSuccess;
+    }
+
+    // Download updated blobs from potential nimbodes
+    public static boolean downloadUpdatedBlob(Map conf, BlobStore blobStore, String key, Set<NimbusInfo> nimbusInfos)
+            throws TTransportException {
+        NimbusClient client = null;
+        ClientBlobStore remoteBlobStore = null;
+        InputStreamWithMeta in = null;
+        AtomicOutputStream out = null;
+        Boolean isSuccess = false;
+        LOG.debug("Download blob NimbusInfos {}", nimbusInfos);
+        for (NimbusInfo nimbusInfo : nimbusInfos) {
+            if (isSuccess) {
+                break;
+            }
+            try {
+                client = new NimbusClient(conf, nimbusInfo.getHost(), nimbusInfo.getPort(), null);
+                remoteBlobStore = new NimbusBlobStore();
+                remoteBlobStore.setClient(conf, client);
+                in = remoteBlobStore.getBlob(key);
+                out = blobStore.updateBlob(key, getNimbusSubject());
+                byte[] buffer = new byte[2048];
+                int len = 0;
+                while ((len = in.read(buffer)) > 0) {
+                    out.write(buffer, 0, len);
+                }
+                if (out != null) {
+                    out.close();
+                }
+                isSuccess = true;
+            } catch (IOException | AuthorizationException exception) {
+                throw new RuntimeException(exception);
+            } catch (KeyNotFoundException knf) {
+                // Catching and logging KeyNotFoundException because, if
+                // there is a subsequent update and delete, the non-leader
+                // nimbodes might throw an exception.
+                LOG.info("KeyNotFoundException {}", knf);
+            } catch (Exception exp) {
+                // Logging an exception while client is connecting
+                LOG.error("Exception {}", exp);
+            }
+        }
+
+        if (!isSuccess) {
+            LOG.error("Could not update the blob with key" + key);
+            return false;
+        }
+        return isSuccess;
+    }
+
+    // Get the list of keys from blobstore
+    public static List<String> getKeyListFromBlobStore(BlobStore blobStore) throws Exception {
+        Iterator<String> keys = blobStore.listKeys();
+        List<String> keyList = new ArrayList<String>();
+        while (keys != null && keys.hasNext()) {
```

Comment:
you are right about short circuit, what i mean is, since it's a while loop, i assume `keys!=null` will be evaluated multiple times, which is redundant because the condition only needs to be evaluated once.